### ISSUE TITLE
False positive: Use snake_case for source file names on all Ruby files

### ISSUE THREAD

**Issue by gosukiwi** [AtomLinter/linter-rubocop#414]
## Describe the bug
Same as #2 Rubocop suggests the filename to be in `snake_case`, but the name is actually using that namin convention. It seems to still be an issue for me.

## Expected behavior
Files with snake_case naming should not be caught by the snake_case cop.

## Actual behavior
Files with snake_case get marked as non-snake-case by the `Style/FileName` cop.

## Environment
 Please complete the following information:

 - Windows 10
 - Atom version 1.47.0
 - linter-rubocop version 2.5.7
 - Ruby version manager rvm (use WSL for running the command `wsl /home/<my-user>/.rvm/gems/ruby-2.6.6/wrappers/rubocop`)
 - Rubocop version 10.84.0

 ```
$ rubocop -V
0.84.0 (using Parser 2.7.1.2, rubocop-ast 0.0.3, running on ruby 2.6.6 x86_64-linux)
```

**Comment by vzamanillo** [comment]
What about running Rubocop against the same file from console? Same results?

**Comment by gosukiwi** [comment]
From console it works as expected.

**Comment by vzamanillo** [comment]
Could you please paste the complete offense message? I can remember I had this issue in the past with the tests but I can't remember how I resolved it.

**Comment by vzamanillo** [comment]
I found it, https://github.com/AtomLinter/linter-rubocop/commit/07c34537c2c27aa6b50073813492de752b33da1e, it says `Rubocop warns about filenames without underscores and with numbers or symbols` it this your case?

**Comment by gosukiwi** [comment]
Thanks for your help @vzamanillo. I tried it with two files: `lib\rubisp\tokenizer\matchers\number.rb` and `spec\rubisp\tokenizer\matchers\number_spec.rb`. It complains about both of them.

I attach an image:

![sc](https://user-images.githubusercontent.com/161972/82926475-9f304a00-9f55-11ea-9e12-b5d5d40847db.png)

**Comment by vzamanillo** [comment]
Are you sure you are running the same Rubocop version from Atom and from console? could you please check the two versions running

- `wsl /home/<my-user>/.rvm/gems/ruby-2.6.6/wrappers/rubocop -V`
- `$(which rubocop) -V` (outside any gemset, from the system default rvm)

please?

Do you have any custom Rubocop configuration via `.rubocop.yml` in the project folder?
Does include the project gemfile the rubocop gem and it is in the bundle? check it with

- `gem list rubocop` 

I can't reproduce your issue but I am working in another environment, that's is the reason of the above questions.

**Comment by gosukiwi** [comment]
It surely is running the same version:

```
# On the Windows side
PS C:\Users\MyUser> wsl /home/<my-user>/.rvm/gems/ruby-2.6.6/wrappers/rubocop -V
0.84.0 (using Parser 2.7.1.2, rubocop-ast 0.0.3, running on ruby 2.6.6 x86_64-linux)
# On the Linux side using WSL 1
/mnt/c/Users/MyUser » $(which rubocop) -V
0.84.0 (using Parser 2.7.1.2, rubocop-ast 0.0.3, running on ruby 2.6.6 x86_64-linux)
```

I have a rubocop configuration file at the root of my project but it's quite minimal:

```yaml
Style/StringLiterals:
  Enabled: true
  EnforcedStyle: double_quotes

Style/LambdaCall:
  EnforcedStyle: braces

Style/Documentation:
  Enabled: false

Layout/LineLength:
  Max: 120

Metrics/BlockLength:
  Enabled: true
  Exclude:
    - spec/**/*
```

I don't have `rubocop` in my Gemfile but I did install it using `gem install`. Running `gem list rubocop` in my project root I get:

```
/mnt/c/Users/MyUser/Workspace/rubisp(master*) » gem list rubocop

*** LOCAL GEMS ***

rubocop (0.84.0)
rubocop-ast (0.0.3)
```

**Comment by vzamanillo** [comment]
Thanks for the info, could you please attach the `linter-rubocop` configuration?

**Comment by gosukiwi** [comment]
Here's my config:

![sc](https://user-images.githubusercontent.com/161972/82960991-6d89a400-9f92-11ea-8ef4-cf6e3aecd314.png)

**Comment by vzamanillo** [comment]
I am not able to reproduce your issue and I can not imagine what could cause it :/ (yet) sorry. I will try to reproduce your environment on a Windows virtual machine, I will keep you informed.

**Comment by gosukiwi** [comment]
Thanks for all your work and help @vzamanillo, I suspect it has something to do with WSL :/

**Comment by vzamanillo** [comment]
Thank you for reporting the issue. It could be, it is the only and unusual "strange agent".

**Comment by gosukiwi** [comment]
Just so we can close this issue, I found the issue. Basically, when using WSL, I was sending Windows line endings to Rubocop, which get escaped, so instead of, for example: `C:\Users\my_file.rb`, Rubocop was getting `C:Usersmy_file.rb`.

I fixed it by creating a `.cmd` file in my PATH to wrap the command, and escape backward slashes to forward slashes:

```
@echo off
REM Transform backward slashes (\) to UNIX forward slashes (/) for all arguments
set "args=%*"
set "args=%args:\=/%"

wsl /home/myusername/.rvm/gems/ruby-2.6.3/wrappers/rubocop %args%
```

Now it works and I don't even have to customize the settings, as it's using the default executable name :)

**Comment by vzamanillo** [comment]
Good catch! Thank you for the info and the workaround.
